frontend/javascripts/oxalis/geometries/skeleton.js

```diff
@@ -485,6 +489,36 @@ class Skeleton {
     });
   }
 
+  /**
+   * Updates a node's position and that of its edges in a WebGL buffer.
+   */
+  updateNodePosition(treeId: number, nodeId: number, position: Vector3, tree: Tree) {
+    const bufferNodeId = this.combineIds(nodeId, treeId);
+    this.update(bufferNodeId, this.nodes, ({ buffer, index }) => {
+      const attribute = buffer.geometry.attributes.position;
+      attribute.set(position, index * 3);
+      return [attribute];
+    });
+
+    const edgePositionUpdater = (edge: Edge, indexOffset: number) => {
+      const bufferEdgeId = this.combineIds(treeId, edge.source, edge.target);
+      this.update(bufferEdgeId, this.edges, ({ buffer, index }) => {
+        const positionAttribute = buffer.geometry.attributes.position;
+        positionAttribute.set(position, index * 6 + indexOffset);
+        return [positionAttribute];
+      });
+    };
+
+    for (const edge of tree.edges.getOutgoingEdgesForNode(nodeId)) {
+      edgePositionUpdater(edge, 0);
+    }
+    for (const edge of tree.edges.getIngoingEdgesForNode(nodeId)) {
+      // The changed node is the target node of the edge which is saved
+      // after the source node in the buffer
```

Comment:
Instead of passing the `indexOffset`, I'd pass a `isIngoingEdge`. Then, `edgePositionUpdater` can do `const indexOffset = isIngoingEdge ? 3 : 0`. Seems like the better abstraction to me.